File: apps/api/internal/domain/link.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type Link struct {
	ID          uuid.UUID  `json:"id" db:"id"`
	UserID      uuid.UUID  `json:"user_id" db:"user_id"`
	FolderID    *uuid.UUID `json:"folder_id" db:"folder_id"`
	URL         string     `json:"url" db:"url"`
	Title       *string    `json:"title" db:"title"`
	Description *string    `json:"description" db:"description"`
	Image       *string    `json:"image" db:"image"`
	CreatedAt   time.Time  `json:"created_at" db:"created_at"`
	UpdatedAt   time.Time  `json:"updated_at" db:"updated_at"`
	// DeletedAt intentionally omitted from API responses
}

// LinkWithTags is the API response shape for a link enriched with its tags.
// Tags is always non-nil (empty slice when no tags) so JSON output is "tags": [].
type LinkWithTags struct {
	Link
	Tags []Tag `json:"tags"`
}

File: apps/web/app/types/link.ts
export interface Link {
  id: string
  url: string
  title: string | null
  description: string | null
  created_at: string
  updated_at: string
  folder_id: string | null
  image: string | null
  user_id: string
  tags: Tag[]
}

export interface Tag {
  id: string
  name: string
  created_at: string
  updated_at: string
  user_id: string
}
